tools/fds/server.cpp

```diff
@@ -168,65 +244,70 @@ int fastdds_discovery_server(
 
     // Retrieve first IP address
     pOp = options[IPADDRESS];
-    if (nullptr == pOp)
+    if (nullptr == pOp && (nullptr == options[XML_FILE] || nullptr != pO_port))
```

Comment:
I am not sure why the last comparison is `!=`. Would you mind explaining?

Reply:
Done in 7d6428651